### ISSUE TITLE
SGF invalid for fechgo league - unknown reasons

### ISSUE THREAD

**Issue by alexmitrani** [climu/openstudyroom#374]
I checked and both players are registered, they tagged the game, komi and time settings are compliant.  I don't know why the system says it is invalid and no explanation is given.  Please help.  Thanks!  

This is the SGF: 

http://files.gokgs.com/games/2019/4/28/Fiona-clachalote.sgf

**Comment by alexmitrani** [comment]
I tried stripping things out to see if there was something in the contents that was causing a problem.  This was also considered invalid: 

(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
RU[Japanese]SZ[19]KM[6.50]TM[1800]OT[5x30 byo-yomi]
PW[Fiona]PB[clachalote]WR[7k]BR[8k]DT[2019-04-28]PC[The KGS Go Server at http://www.gokgs.com/]
RE[W+21.50]
;B[qd]BL[1779.066]
;W[od]WL[1770.578]
;B[pf]BL[1772.265]
;W[qc]WL[1759.696]
;B[rc]BL[1762.26]
;W[pc]WL[1754.056]
;B[pp]BL[1750.265]
;W[dp]WL[1688.015]
;B[dd]BL[1736.648]
;W[fc]WL[1682.308]
;B[cf]BL[1729.561]
;W[nq]WL[1667.449]
;B[oq]BL[1720.529]
;W[np]WL[1662.237]
;B[qn]BL[1718.916]
;W[jq]WL[1655.089]
;B[cn]BL[1715.454]
;W[co]WL[1647.015]
;B[dn]BL[1713.919]
;W[fp]WL[1641.974]
;B[cj]BL[1711.758]
;W[db]WL[1629.17]
;B[cc]BL[1705.466]
;W[ic]WL[1621.502]
;B[qj]BL[1678.508]
;W[nf]WL[1592.319]
;B[pd]BL[1654.466]
;W[nc]WL[1583.583]
;B[jn]BL[1640.684]
;W[hp]WL[1510.477]
;B[fn]BL[1632.099]
;W[ej]WL[1496.761]
;B[fe]BL[1629.285]
;W[ie]WL[1454.382]
;B[ei]BL[1623.525]
;W[dj]WL[1420.3]
;B[di]BL[1620.318]
;W[fj]WL[1364.811]
;B[fi]BL[1607.672]
;W[gj]WL[1335.743]
;B[dk]BL[1592.914]
;W[gi]WL[1317.826]
;B[gh]BL[1590.041]
;W[hh]WL[1276.165]
;B[hg]BL[1573.21]
;W[ci]WL[1187.4]
;B[ck]BL[1568.281]
;W[fh]WL[1180.354]
;B[fg]BL[1548.9]
;W[eh]WL[1160.384]
;B[dh]BL[1542.594]
;W[eg]WL[1098.766]
;B[gg]BL[1513.084]
;W[ih]WL[787.677]
;B[ig]BL[1498.885]
;W[jh]WL[778.314]
;B[jg]BL[1493.593]
;W[kh]WL[776.113]
;B[kg]BL[1490.242]
;W[hn]WL[762.063]
;B[nn]BL[1484.426]
;W[ln]WL[753.713]
;B[go]BL[1482.178]
;W[ho]WL[751.955]
;B[gp]BL[1480.242]
;W[gq]WL[749.965]
;B[mo]BL[1473.739]
;W[lp]WL[744.312]
;B[lo]BL[1470.488]
;W[ko]WL[741.723]
;B[kn]BL[1467.789]
;W[lm]WL[736.422]
;B[kp]BL[1465.537]
;W[jo]WL[733.441]
;B[mp]BL[1463.402]
;W[lq]WL[724.767]
;B[mq]BL[1462.088]
;W[kq]WL[723.043]
;B[mr]BL[1460.248]
;W[lj]WL[673.255]
;B[hd]BL[1452.385]
;W[hc]WL[660.805]
;B[id]BL[1450.502]
;W[jd]WL[658.723]
;B[je]BL[1448.875]
;W[kd]WL[654.769]
;B[ke]BL[1447.129]
;W[ld]WL[652.654]
;B[le]BL[1443.923]
;W[qh]WL[592.155]
;B[ph]BL[1439.375]
;W[qi]WL[581.649]
;B[pi]BL[1436.723]
;W[rj]WL[539.417]
;B[qk]BL[1433.572]
;W[rk]WL[531.229]
;B[ql]BL[1426.297]
;W[qg]WL[528.732]
;B[pg]BL[1421.535]
;W[re]WL[522.296]
;B[qf]BL[1419.409]
;W[rf]WL[517.855]
;B[rd]BL[1414.49]
;W[bo]WL[467.349]
;B[bn]BL[1411.095]
;W[an]WL[464.934]
;B[am]BL[1400.303]
;W[ao]WL[463.267]
;B[bl]BL[1399.008]
;W[fm]WL[460.266]
;B[gm]BL[1395.606]
;W[gn]WL[316.484]
;B[fo]BL[1392.285]
;W[ep]WL[306.068]
;B[em]BL[1344.137]
;W[fl]WL[291.047]
;B[cb]BL[1304.326]
;W[lr]WL[283.308]
;B[eb]BL[1301.065]
;W[ms]WL[280.581]
;B[ns]BL[1298.635]
;W[ls]WL[274.955]
;B[nr]BL[1297.168]
;W[ec]WL[255.204]
;B[dc]BL[1294.382]
;W[fb]WL[254.136]
;B[ij]BL[1261.167]
;W[hi]WL[208.841]
;B[sh]BL[1258.607]
;W[rh]WL[103.206]
;B[rl]BL[1254.28]
;W[si]WL[91.501]
;B[sk]BL[1250.281]
;W[sg]WL[30]OW[5]
;B[lh]BL[1236.359]
;W[li]WL[30]OW[5]
;B[mh]BL[1234.647]
;W[gd]WL[30]OW[5]
;B[he]BL[1231.07]
;W[ge]WL[30]OW[5]
;B[hf]BL[1220.174]
;W[dg]WL[30]OW[5]
;B[ch]BL[1209.248]
;W[cg]WL[30]OW[5]
;B[bh]BL[1201.698]
;W[bg]WL[30]OW[5]
;B[bi]BL[1199.666]
;W[ee]WL[30]OW[5]
;B[ff]BL[1154.292]
;W[ed]WL[30]OW[5]
;B[ef]BL[1150.62]
;W[de]WL[30]OW[5]
;B[df]BL[1148.091]
;W[ce]WL[30]OW[5]
;B[be]BL[1142.997]
;W[bd]WL[30]OW[5]
;B[bf]BL[1137.012]
;W[cd]WL[30]OW[5]
;B[bc]BL[1135.698]
;W[ad]WL[30]OW[5]
;B[da]BL[1126.936]
;W[nl]WL[30]OW[4]
;B[mm]BL[1122.779]
;W[ml]WL[30]OW[4]
;B[ll]BL[1120.442]
;W[km]WL[30]OW[4]
;B[kl]BL[1114.448]
;W[jm]WL[30]OW[4]
;B[jl]BL[1099.179]
;W[lk]WL[30]OW[4]
;B[mn]BL[1094.658]
;W[in]WL[30]OW[4]
;B[mk]BL[1089.342]
;W[nj]WL[30]OW[4]
;B[mi]BL[1087.651]
;W[mj]WL[30]OW[4]
;B[ok]BL[1085.972]
;W[nk]WL[30]OW[4]
;B[el]BL[1081.277]
;W[fk]WL[30]OW[4]
;B[qb]BL[1066.952]
;W[pb]WL[30]OW[4]
;B[rb]BL[1063.862]
;W[om]WL[30]OW[4]
;B[on]BL[1061.22]
;W[pm]WL[30]OW[4]
;B[ne]BL[1055.218]
;W[oe]WL[30]OW[4]
;B[md]BL[1053.739]
;W[mc]WL[30]OW[4]
;B[of]BL[1050.497]
;W[qm]WL[30]OW[3]
;B[rm]BL[1045.89]
;W[pn]WL[30]OW[3]
;B[po]BL[1034.688]
;W[ni]WL[30]OW[3]
;B[nh]BL[1031.799]
;W[oj]WL[30]OW[3]
;B[pj]BL[1026.692]
;W[ae]WL[30]OW[3]
;B[af]BL[1019.498]
;W[fa]WL[30]OW[3]
;B[ea]BL[1016.886]
;W[ba]WL[30]OW[3]
;B[bb]BL[993.141]
;W[ac]WL[30]OW[3]
;B[ab]BL[986.465]
;W[nd]WL[30]OW[3]
;B[me]BL[982.227]
;W[do]WL[30]OW[3]
;B[en]BL[979.26]
;W[qa]WL[30]OW[3]
;B[ra]BL[974.9]
;W[pa]WL[30]OW[3]
;B[ol]BL[969.185]
;W[nm]WL[30]OW[3]
;B[pl]BL[967.11]
;W[jj]WL[30]OW[3]
;B[oi]BL[963.512]
;W[sj]WL[30]OW[3]
;B[sl]BL[958.07]
;W[sd]WL[30]OW[3]
;B[sc]BL[956.359]
;W[se]WL[30]OW[3]
;B[fd]BL[953.443]
;W[jc]WL[30]OW[3]
;B[gf]BL[948.158]
;W[gc]WL[30]OW[3]
;B[]BL[948.158]
;W[]WL[30]OW[3])

**Comment by alexmitrani** [comment]
I was expecting it to say that it was invalid because the tag was missing but it didn't.  It just continued to say that it was invalid without any explanation.

**Comment by alexmitrani** [comment]
It's been scraped now.  I don't know what happened there.

**Comment by alexmitrani** [comment]
Maybe when I tried uploading it manually it was already in the queue to be scraped.  If that is the case the manual upload might be treated as invalid to avoid the game being counted twice.  If this is so, it would be great if there could be a message to the user explaining the situation.  Thanks.

**Comment by climu** [comment]
It seems there is an issue here and I will try to work on this before next week.

In theory when a  sgf is in queue to be scraped, you can upload the same one without worry. The one in the queue will then be scraped with the output "same sgf already in db".

There shouldn't be an invalid sgf without any error message. As I said I ll work on that.

**Comment by climu** [comment]
Bad news. I won't be able to work on that for now.

Since the automatic scraper seems to work I guess it's not that urgent.

**Comment by alexmitrani** [comment]
I had a similar experience just now, it was saying a kifu was invalid without any explanation.  I found a possible cause and a workaround.  In Django-Admin the SGF was on the list and marked as invalid.  I deleted that record.  Then I went back to the admin SGF interface and scrobbled the kifu anew, this time it worked.  So my tentative conclusion is that the case where someone tries to scrobble a pre-existing SGF needs to be dealt with better, maybe with an error message and / or an option to replace the pre-existing SGF.  
Thanks.
